omegaconf/base.py

```diff
@@ -568,7 +540,15 @@ def quoted_string_callback(quoted_str: str) -> str:
             resolver_interpolation_callback=resolver_interpolation_callback,
             quoted_string_callback=quoted_string_callback,
         )
-        return visitor.visit(parse_tree)
+        try:
+            return visitor.visit(parse_tree)
+        except InterpolationResolutionError:
+            raise
+        except Exception as exc:
+            # Other kinds of exceptions are wrapped in an `InterpolationResolutionError`.
+            raise InterpolationResolutionError(
+                f"{type(exc).__name__} raised while resolving interpolation: {exc}"
+            )
```

Comment:
I think at least chaining the exception here is important. see what it looks like with combinations of retaining the stack trace and chaining the exception.

Reply:
Also done in 5718bf0 (although again, chaininig didn't seem to make any difference, so I didn't do it)